core/thread/inc/ROOT/TThreadedObject.hxx

```diff
@@ -153,12 +153,18 @@ namespace ROOT {
       /// objects.
       /// \tparam ARGS Arguments of the constructor of T
       template<class ...ARGS>
-      TThreadedObject(ARGS&&... args): fObjPointers(fgMaxSlots, nullptr)
+      TThreadedObject(ARGS&&... args)
       {
          fDirectories = Internal::TThreadedObjectUtils::DirCreator<T>::Create(fgMaxSlots);
 
          TDirectory::TContext ctxt(fDirectories[0]);
          fModel.reset(Internal::TThreadedObjectUtils::Detacher<T>::Detach(new T(std::forward<ARGS>(args)...)));
+
+         if(ROOT::IsImplicitMTEnabled()) {
+            TThreadedObject<T>::fgMaxSlots =  ROOT::GetImplicitMTPoolSize();
```

Comment:
Why change the value for each TThreadObject creation?

Also, in order for this line (and the next) to be thread safe fgMaxSlots needs to be an std::atomic.  [Otherwise, per the standard, the behavior when the reading (line 167) and the writing (line 164) happen at the 'same time' is undefined]

Reply:
Good point. Will look into it 🤔

Reply:
@pcanal doesn't everything go bonkers anyway if users call `EnableImplicitMT` in a thread while they are using implicit-mt-aware features in other threads concurrently?